### PR TITLE
change mapper interface method names to generic names

### PR DESCRIPTION
**Pull request by jCube1980**
closes https://github.com/jhipster/generator-jhipster/issues/5586

Travis [build](https://travis-ci.org/jCube1980/generator-jhipster/builds/224716153) is successful and generated apps work fine.

### CODE REVIEW

File: generators/entity/templates/server/src/main/java/package/service/mapper/_BaseEntityMapper.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+-%>
+package <%=packageName%>.service.mapper;
+
+import java.util.List;
+
+/**
+ * Contract for a generic entity to dto mapper.
+ */
+
+public interface EntityMapper <D, E> {
```
Comment (gzsombor) [inline generators/entity/templates/server/src/main/java/package/service/mapper/_BaseEntityMapper.java]: I prefer to have a readable type arguments, eg : EntityMapper<DTO, ENTITY>

**Comment by gzsombor** [comment]
Other than this one minor issue, it's LGTM +1

File: generators/entity/templates/server/src/main/java/package/service/mapper/_BaseEntityMapper.java (hunk 1 above)
Comment (jCube1980) [inline generators/entity/templates/server/src/main/java/package/service/mapper/_BaseEntityMapper.java]: I fully agree with the intention of keeping the type names readable/descriptive. 

I followed the Sun/Oracle [convention](http://stackoverflow.com/questions/2900881/generic-type-parameter-naming-convention-for-java-with-multiple-chars) which recommends "single, uppercase letters for type parameters". This helps developers effortlessly distinguish the difference between a type variable and an ordinary class or interface name. Also, this convention guarantees consistency. 

Another option I came across is adding details in the javadoc comments section. This approach follows the convention (thereby removing the possible confusion) and adds the detail you wanted. 

```
/**
 Contract for a generic dto to entity mapper.
 @param <D> - DTO
 @param <E> - Entity
 */
public interface EntityMapper <D, E> {
```

If it doesn't look good, I can change it as per your suggestion.

**Comment by jCube1980** [comment]
Hi @gzsombor, I have changed the generic type parameter names as you had suggested. However the Travis build is failing due to a timeout issue. Any chance to re-trigger just the two steps that are failing? I re-triggered the failing steps in my fork and it completed successfully.

**Comment by pascalgrimaud** [comment]
I'm restarting the 2 jobs

**Comment by jCube1980** [comment]
The 2 jobs failed again.. :(

**Pull request merged**